storage/innobase/xtrabackup/test/t/xbcloud.sh

```diff
@@ -165,8 +169,111 @@ if  grep -q "Upload failed" $topdir/pxb-2279.log ; then
     die 'xbcloud exit with error on upload'
 fi
 
+# PXB-2112 - Support for storage class
+test_storage_class=false
+
+if grep -q "storage=s3" $topdir/xbcloud.cnf ; then
+  vlog "Testing S3 storage class"
+  test_storage_class=true
+  storage_class_parameter="--s3-storage-class=GLACIER"
+elif grep -q "storage=google" $topdir/xbcloud.cnf ; then
+  vlog "Testing GC storage class"
+  test_storage_class=true
+  storage_class_parameter="--google-storage-class=COLDLINE"
+fi
+
+if grep -q "minioadmin" $topdir/xbcloud.cnf ; then
+  test_storage_class=false
+fi
+
+if [ "$test_storage_class" = true ]; then
+  run_cmd touch $topdir/xtrabackup_info
+  cd $topdir/
+  xbstream -c xtrabackup_info | \
+  xbcloud --defaults-file=$topdir/xbcloud.cnf ${storage_class_parameter} \
+   put ${storage_class_folder} 2> $topdir/storage_class.log
+
+ if  grep -q "expected objects using storage class" $topdir/storage_class.log ; then
+    xbcloud --defaults-file=$topdir/xbcloud.cnf delete \
+     ${storage_class_folder}
+     die 'xbcloud did not upload using correct storage class'
+ fi
+ if  grep -q "Upload failed" $topdir/storage_class.log ; then
+   xbcloud --defaults-file=$topdir/xbcloud.cnf delete \
+     ${storage_class_folder}
+     die 'xbcloud exit with error on upload'
+ fi
+ run_cmd xbcloud --defaults-file=$topdir/xbcloud.cnf delete \
+  ${storage_class_folder}
+fi
+
 # cleanup
 run_cmd xbcloud --defaults-file=$topdir/xbcloud.cnf delete \
 	${full_backup_name} --parallel=4
 run_cmd xbcloud --defaults-file=$topdir/xbcloud.cnf delete \
 	${inc_backup_name} --parallel=4
+rm -rf ${full_backup_dir}
+rm -rf $topdir/downloaded_full
+
+
+#check if password is stored in backup directory or in ps output
+
+## grep for all key in secret_key
+params=($(echo $XBCLOUD_CREDENTIALS | tr ";" "\n"))
+secret_keys=();
+for param in "${params[@]}"
+do
+ if [[ $param =~ "key=" ]]; then
```

Comment:
Will this also catch `azure-storage-account` ?

Reply:
no. it will just grep keys. let me fixed that.